src/codeGenerators/contract/solidity/toContract.ts

```diff
@@ -69,11 +69,24 @@ function codeGenerator(node: any) {
         node.isConstructor ? 'constructor ' : 'function '
       }${node.name} (${codeGenerator(node.parameters)}) ${node.visibility} {`;
       const body = codeGenerator(node.body);
+      let returnStatement = ` `;
+      if(node.returnParameters.parameters) {
+      node.returnParameters.parameters.forEach( node => {
+        if(node.name && node.isSecret === true)
+         returnStatement = `  `;
+        else if(node.name && node.isSecret === false)
+         returnStatement = `return ${node.name} `;
+        else if(node.value)
+        returnStatement = `return ${node.value} `;
+        });
+    }
```

Comment:
This part fails for me for two reasons I can see:

- there is no `;` to end the line, so the contract doesn't compile
- there is no return declaration in fn signature (e.g. the returns part in`function add(secret uint256 value) public returns (uint256)`)